contracts/JBTieredLimitedNFTRewardDataSource.sol

```diff
@@ -338,72 +302,88 @@ contract JBTieredLimitedNFTRewardDataSource is
     @notice 
     Mints a token for a given contribution to the beneficiary.
 
+    @dev
+    `_data.metadata` should include the number of rewards being requested in bits 32-39.
+
     @param _data The Juicebox standard project contribution data.
   */
   function _processContribution(JBDidPayData calldata _data) internal override {
     // Make the contribution is being made in the expected token.
     if (_data.amount.token != contributionToken) return;
 
-    // Keep a reference to the token ID.
-    (uint256 _tokenId, uint256 _tierNumber) = _generateTokenId(_data.amount.value);
+    // Keep a reference to the number of rewards being processed. Skip the first 32 bits which are used by the JB protocol.
+    uint256 _numRewards = uint256(uint8(_bytesToUint(_data.metadata) << 32));
 
-    // Make sure there's a token ID.
-    if (_tokenId == 0)
-      if (_tierNumber > 0)
-        // If there's no token ID but there is a tier, the tier must have sold out.
-        revert NOT_AVAILABLE();
-      // If there is no tier, accept the payment normally without minting.
-      else return;
+    // Keep a reference to the value remaining.
+    uint256 _remainingValue = _data.amount.value;
 
-    // If there's a token to mint, do so and increment the tier supply.
-    _mint(_data.beneficiary, _tokenId);
+    // Keep a reference to the tier ID being iterated on.
+    uint256 _tierId;
 
-    emit Mint(_tokenId, _tierNumber, _data.beneficiary, _data.amount.value, msg.sender);
-  }
+    // Keep a reference to the token ID being iterated on.
+    uint256 _tokenId;
 
-  /** 
-    @notice
-    Finds the token ID and tier given a contribution amount. 
+    for (uint256 _i; _i < _numRewards; ) {
+      // Tier IDs are in chuncks of 32 bits starting in bit 40.
+      _tierId = uint256(uint32(_bytesToUint(_data.metadata) << (40 + 32 * _i)));
 
-    @param _amount The amount of the contribution.
+      // If no tier specified, accept the funds without minting anything.
+      if (_tierId == 0) continue;
```

Comment:
should we use this as a fallback (and then mint the highest floor possible, like in prev version)?

Reply:
yeah i like this. There could be a constructor flag `shouldMintDefaultTier`, if on, a payment coming in will try to mint a the highest tier it can.